src/main/java/carsharing/customer/CustomerDAOImpl.java

```diff
@@ -0,0 +1,214 @@
+package carsharing.customer;
+
+import carsharing.car.Car;
+import carsharing.company.Company;
+import carsharing.configuration.DBManager;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+
+public class CustomerDAOImpl extends DBManager implements CustomerDAO {
+
+    private static final String SELECT_NAME_BY_CUSTOMER = "SELECT name from customer ORDER BY ID";
+    private static final String SELECT_NAME_AND_CAR_ID = "SELECT name, rented_car_id FROM customer WHERE id = ?";
+    private static final String SELECT_NAME_AND_COMPANY_ID = "SELECT name, company_id FROM car WHERE id = ?";
+    private static final String SELECT_NAME_BY_COMPANY = "SELECT name FROM company WHERE id = ?";
+    private static final String UPDATE_CUSTOMER_WITH_NULL_RENTED_CAR = "UPDATE customer SET rented_car_id = NULL WHERE id = ?";
+    private static final String SELECT_ID_BY_CAR_NAME = "SELECT id FROM car WHERE name = ?";
+    private static final String UPDATE_CUSTOMER = "UPDATE customer SET rented_car_id = ? WHERE id = ?";
+    private static final String SELECT_ID_BY_COMPANY = "SELECT id FROM company WHERE name = ?";
+    private static final String SELECT_RENTED_CAR = "SELECT rented_car_id FROM customer WHERE rented_car_id IS NOT NULL";
+    private static final String SELECT_ID_NAME_BY_CAR = "SELECT id, name FROM car WHERE company_id = ? ORDER BY id";
+    private static final String INSERT_INTO_CUSTOMER_BY_NAME = "INSERT INTO customer (name) VALUES ?";
+
+    private final DBManager DBManager;
+    private Set<Integer> rentedCars;
```

Comment:
Why do we need store state outside of DB?

Reply:
This is car id, we need it to assign a car id.
I fix this